NitroxClient/GameLogic/TimeManager.cs

```diff
@@ -0,0 +1,37 @@
+using NitroxModel.Packets;
+using System;
+
+namespace NitroxClient.GameLogic;
+
+public class TimeManager
+{
+    /// <summary>
+    /// Latest moment at which we updated the time
+    /// </summary>
+    private DateTimeOffset latestRegistrationTime;
+    /// <summary>
+    /// Latest registered value of the time
+    /// </summary>
+    private double latestRegisteredTime;
+
+    public double CurrentTime
```

Comment:
They should probably never. During the same frame, we don't want a different time to be recalculated. So you'll just need to take the current DayNightCycle.main.timePassed(...) instead.
Actually it depends on the use case, but most of Subnautica's code is made to work like this and we don't want to modify this behaviour.
Although I'm starting to think about changing this property and the current way of recalculating time to instead modify DayNightCycle.deltaTime property (which would be more relevant considering that it's used in other places)

Reply:
This time property could be useful for purposes which I can't think of for now so I guess I can at least add some comment here explaining what I commented earlier.

Update on what I said earlier: I won't change daynightcycle.deltaTime because it's used by electricity generators (as an example) and if you skip a day or so, you don't want your generator to think that the passed time is of one day.